csharp/src/Google.Protobuf.Test/CodedInputStreamAsyncTest.cs

```diff
@@ -0,0 +1,502 @@
+#region Copyright notice and license
```

Comment:
If only C# supported macros, a macro/template would definitely work here. However c# does not support macros/templates - so basically it is impossible to introduce sync/async polymorphism.